src/AggregateFunctions/IAggregateFunction.h

```diff
@@ -125,7 +125,7 @@ class IAggregateFunction : public std::enable_shared_from_this<IAggregateFunctio
     }
 
     /// It is not necessary to delete data.
-    virtual bool hasTrivialDestructor() const = 0;
+    constexpr virtual bool hasTrivialDestructor() const = 0;
```

Comment:
So it can be evaluated at compile time and produce better code if possible. For example if you do:
```
    if (!func->hasTrivialDestructor())
        func->destroy(val);
```

And the type of func is known at compile time, then the compiler can (and most likely will if you use `if constexpr()`) remove that if call. This is mostly useful for templates but it could be removed as currently it's not being made advantage of, as both calls are done via pointers to the base class.